includes/customizer/class-placement-customize-control.php

```diff
@@ -0,0 +1,351 @@
+	 */
+	private $placement = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WP_Customize_Manager $manager Customizer manager.
+	 * @param string               $id      Control ID.
+	 * @param array                $args    Control arguments.
+	 */
+	public function __construct( $manager, $id, $args = [] ) {
+		// Available ad units.
+		$this->providers = Providers::get_active_providers_data();
+		$this->bidders   = \Newspack_Ads\get_bidders();
+
+		// Placement configuration.
+		if ( isset( $args['placement'] ) ) {
+			$placements      = Placements::get_placements();
+			$this->placement = $placements[ $args['placement'] ];
+		} else {
+			return new \WP_Error( 'newspack_ads_placement_customize_control_no_placement', __( 'No placement specified.', 'newspack-ads' ) );
+		}
+
+		parent::__construct( $manager, $id, $args );
+	}
+
+	/**
+	 * To Json.
+	 */
+	public function to_json() {
+		parent::to_json();
+
+		$value = $this->value();
+		if ( $value ) {
+			if ( $this->placement['hook_name'] ) {
+				$this->json['provider'] = $this->get_provider_value();
+				$this->json['ad_unit']  = $this->get_ad_unit_value();
+			}
+			if ( isset( $this->placement['hooks'] ) && count( $this->placement['hooks'] ) ) {
+				foreach ( array_keys( $this->placement['hooks'] ) as $hook_key ) {
+					$this->json['hooks'][ $hook_key ]['provider'] = $this->get_provider_value( $hook_key );
+					$this->json['hooks'][ $hook_key ]['ad_unit']  = $this->get_ad_unit_value( $hook_key );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Get the value of the provider given its hook key.
+	 *
+	 * @param string $hook_key Optional hook key, will look root placement otherwise.
+	 *
+	 * @return string Provider ID or \Newspack_Ads\Providers::DEFAULT_PROVIDER if not found.
+	 */
+	private function get_provider_value( $hook_key = '' ) {
+		$value   = json_decode( $this->value(), true );
+		$default = Providers::DEFAULT_PROVIDER;
+		$data    = $value;
+		if ( ! empty( $hook_key ) && isset( $value['hooks'], $value['hooks'][ $hook_key ] ) ) {
+			$data = $value['hooks'][ $hook_key ];
+		}
+		return isset( $data['provider'] ) ? $data['provider'] : $default;
+	}
+
+	/**
+	 * Get the value of an ad unit given its hook key.
+	 *
+	 * @param string $hook_key Optional hook key, will look root placement otherwise.
+	 *
+	 * @return string Ad unit ID or empty string if not found.
+	 */
+	private function get_ad_unit_value( $hook_key = '' ) {
+		$value = json_decode( $this->value(), true );
+		$data  = $value;
+		if ( ! empty( $hook_key ) && isset( $value['hooks'], $value['hooks'][ $hook_key ] ) ) {
+			$data = $value['hooks'][ $hook_key ];
+		}
+		return isset( $data['ad_unit'] ) ? $data['ad_unit'] : '';
+	}
+
+	/**
+	 * Get the value of "stick to top" feature given its hook key.
+	 *
+	 * @param string $hook_key Optional hook key, will look root placement otherwise.
+	 *
+	 * @return string Ad unit ID or empty string if not found.
+	 */
+	private function get_stick_to_top_value( $hook_key = '' ) {
```

Comment:
Nice catch! Since "stick to top" is a container class, 57e77e7e94e4325f6a6fb02cf15661106c460681 tweaks the preview a bit to inject placement data into the HTML and uses the preview javascript to toggle its container class.